### PR TITLE
feat: add Spark-compatible approx_percentile / percentile_approx

### PR DESCRIPTION
**Pull request by davidlghellin**
Replace the direct mapping to DataFusion's `approx_percentile_cont` with a new `SparkApproxPercentile` UDAF that matches Spark semantics:

- Accept an array of percentiles, returning `array<input_type>` (the previous mapping errored on the array form); this fixes the `approx_percentile` and `percentile_approx` doctests in the Spark 3.5.7 suite.
- Preserve the input type (int->int, bigint->bigint, decimal->decimal, float/double, and ANSI interval/duration) instead of always returning double.
- Select percentiles with the discrete nearest-rank method (x[ceil(q*N)-1], q=0 -> min), the value Spark's percentile_approx converges to, verified against the Spark JVM.
- Use `Signature::user_defined` + `coerce_types` to validate arity (2-3 args) and input type at planning time; previously a 4th argument was silently accepted instead of erroring like Spark.

Register the UDAF for both the aggregate and window paths and in the distributed execution codec. Add a feature test covering numeric, decimal, interval/duration, array, NULL, group-by and argument-validation cases. Two scenarios are marked @sail-bug for the Sail-wide interval subrange widening (`interval month` -> `interval year to month`), which affects every interval-returning expression and is unrelated to this function.

### CODE REVIEW

File: crates/sail-function/src/aggregate/approx_percentile.rs [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+                offsets,
+                Arc::new(values_array),
+                None,
+            );
+            Ok(ScalarValue::List(Arc::new(list_array)))
+        } else {
+            let value = calculate_percentile_disc::<T>(values, self.percentiles[0]);
+            ScalarValue::new_primitive::<T>(value, &self.data_type)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (davidlghellin) [reply crates/sail-function/src/aggregate/approx_percentile.rs:494]: Already addressed — `evaluate` now calls `calculate_percentiles_disc`, which sorts the values once and indexes per percentile (no per-percentile clone, no repeated `select_nth_unstable_by`). The nearest-rank index formula lives in the shared `percentile_disc_index` helper in `aggregate/utils.rs` so it isn't duplicated here.
